src/resource.cc

```diff
@@ -10,4 +10,22 @@ void Resource::BumpStateId() {
   nextstate_id_++;
 }
 
+std::vector<Resource::Ptr> Resource::Package(Package::Ptr pkg) {
+  std::vector<Resource::Ptr> rs_pkgd;
+  Resource::Ptr r_pkgd;
+
+  double fill_mass = pkg->GetFillMass(quantity());
+  if (fill_mass == 0) {
+    return rs_pkgd;
+  }
+
+  while (quantity() > pkg->fill_min()) {
+    double pkg_fill = std::min(quantity(), fill_mass);
+    r_pkgd = ExtractRes(pkg_fill);
```

Comment:
This method is explicitly undefined for `Resource` objects.  I don't think we can/should have a `Package` method either.

See comment in the header.